File: C,C++/p299.c
#include <stdio.h>

int main () {
	int kasus, i;
	scanf("%d", &kasus);
	if (kasus == 0) {
		printf("Optimal train swapping takes 0 swaps.\n");
	}
	for (i = 0; i < kasus; i++) {
		int banyak, j;
		scanf("%d", &banyak);
		if (banyak == 0) {
			printf("Optimal train swapping takes 0 swaps.\n");
			continue;
		}
		int data [banyak];
		for (j = 0; j < banyak; j++) {
			scanf("%d", &data[j]);
		}
		
		int status = 0, res = 0;
		
		while (1) {
			status = 0;
			for (j = 0; j < banyak-1; j++) {
				if (data[j] > data[j+1]) {
					int temp = data[j];
					data[j] = data[j+1];
					data[j+1] = temp;
					status = 1;
					res++;
				}
			}
			
			if (status == 0) break;
		}
		
		printf("Optimal train swapping takes %d swaps.\n", res);
	}
	return 0;
}

File: Java/p299.java

import java.util.Scanner;

public class p299 {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int kasus = input.nextInt();
        if (kasus == 0) {
            System.out.println("Optimal train swapping takes 0 swaps.");
        }
        for (int i = 0; i < kasus; i++) {
            int banyak = input.nextInt();
            if (banyak == 0) {
                System.out.println("Optimal train swapping takes 0 swaps.");
                continue;
            }
            int[] data = new int[banyak];
            for (int j = 0; j < banyak; j++) {
                data[j] = input.nextInt();
            }

            int res = 0;
            boolean status;

            while (true) {
                status = false;

                for (int j = 0; j < banyak - 1; j++) {
                    if (data[j] > data[j + 1]) {
                        int temp = data[j];
                        data[j] = data[j + 1];
                        data[j + 1] = temp;
                        res++;
                        status = true;
                    }
                }

                if (!status) {
                    break;
                }
            }

            System.out.printf("Optimal train swapping takes %d swaps.\n", res);
        }
    }
}